src/frontend/screens/Library/components/RecentlyPlayed/index.tsx

```diff
@@ -1,44 +1,65 @@
 import React, { useContext, useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { getRecentGames } from 'frontend/helpers/library'
 import ContextProvider from 'frontend/state/ContextProvider'
-import { GameInfo, GameStatus, Runner } from 'common/types'
+import { AppSettings, GameInfo, RecentGame, Runner } from 'common/types'
 import { GamesList } from '../GamesList'
+import { configStore } from 'frontend/helpers/electronStores'
 
 interface Props {
   handleModal: (appName: string, runner: Runner) => void
   onlyInstalled: boolean
 }
 
+function getRecentGames(libraries: GameInfo[], limit: number): GameInfo[] {
+  const recentGames =
+    (configStore.get('games.recent', []) as Array<RecentGame>) || []
+
+  const games: GameInfo[] = []
+
+  for (const recent of recentGames) {
+    const found = libraries.find(
+      (game: GameInfo) => game.app_name === recent.appName
+    )
+    if (found) {
+      games.push(found)
+      if (games.length === limit) break
```

Comment:
since the list of recent games can grow a lot, stop the loop once we have enough games to show